<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProjectExplorer::Internal::ProjectExplorerSettingsPageUi</class>
 <widget class="QWidget" name="ProjectExplorer::Internal::ProjectExplorerSettingsPageUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>831</width>
    <height>507</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QGroupBox" name="directoryGroupBox">
     <property name="title">
      <string>Projects Directory</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <property name="labelAlignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <item row="0" column="0" colspan="2">
       <widget class="QRadioButton" name="currentDirectoryRadioButton">
        <property name="text">
         <string>Current directory</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">directoryButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="directoryRadioButton">
        <property name="text">
         <string>Directory</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">directoryButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="Utils::PathChooser" name="projectsDirectoryPathChooser"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="buildAndRunGroupBox">
     <property name="title">
      <string>Build and Run</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="saveAllFilesCheckBox">
        <property name="text">
         <string>Save all files before build</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="cleanOldAppOutputCheckBox">
        <property name="text">
         <string>Clear old application output on a new run</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="buildProjectBeforeDeployCheckBox">
        <property name="text">
         <string>Always build project before deploying it</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="mergeStdErrAndStdOutCheckBox">
        <property name="toolTip">
         <string>Enabling this option ensures that the order of interleaved messages from stdout and stderr is preserved, at the cost of disabling highlighting of stderr.</string>
        </property>
        <property name="text">
         <string>Merge stderr and stdout</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="deployProjectBeforeRunCheckBox">
        <property name="text">
         <string>Always deploy project before running it</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="wrapAppOutputCheckBox">
        <property name="text">
         <string>Word-wrap application output</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="showCompileOutputCheckBox">
        <property name="text">
         <string>Open Compile Output pane when building</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Limit application output to </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="maxAppOutputBox">
           <property name="minimum">
            <number>500</number>
           </property>
           <property name="maximum">
            <number>1000000</number>
           </property>
           <property name="singleStep">
            <number>500</number>
           </property>
           <property name="value">
            <number>100000</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>lines</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="showRunOutputCheckBox">
        <property name="text">
         <string>Open Application Output pane on output when running</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="showDebugOutputCheckBox">
        <property name="text">
         <string>Open Application Output pane on output when debugging</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QCheckBox" name="promptToStopRunControlCheckBox">
        <property name="toolTip">
         <string>Asks before terminating the running application in response to clicking the stop button in Application Output.</string>
        </property>
        <property name="text">
         <string>Always ask before stopping applications</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="labelStopBeforeBuild">
          <property name="text">
           <string>Stop applications before building:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="stopBeforeBuildComboBox">
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Same Project</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>All</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="8" column="0" colspan="2">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="jomCheckbox">
          <property name="whatsThis">
           <string/>
          </property>
          <property name="text">
           <string>Use jom instead of nmake</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="jomLabel">
          <property name="text">
           <string>&lt;i&gt;jom&lt;/i&gt; is a drop-in replacement for &lt;i&gt;nmake&lt;/i&gt; which distributes the compilation process to multiple CPU cores. The latest binary is available at &lt;a href=&quot;http://download.qt.io/official_releases/jom/&quot;&gt;http://download.qt.io/official_releases/jom/&lt;/a&gt;. Disable it if you experience problems with your builds.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="indent">
           <number>20</number>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="9" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="topMargin">
         <number>12</number>
        </property>
        <item>
         <widget class="QLabel" name="buildDirLabel">
          <property name="text">
           <string>Default build directory:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="buildDirectoryEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="resetButton">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
     <zorder>saveAllFilesCheckBox</zorder>
     <zorder>buildProjectBeforeDeployCheckBox</zorder>
     <zorder>deployProjectBeforeRunCheckBox</zorder>
     <zorder>showCompileOutputCheckBox</zorder>
     <zorder>cleanOldAppOutputCheckBox</zorder>
     <zorder>mergeStdErrAndStdOutCheckBox</zorder>
     <zorder>wrapAppOutputCheckBox</zorder>
     <zorder>widget</zorder>
     <zorder>promptToStopRunControlCheckBox</zorder>
     <zorder>showRunOutputCheckBox</zorder>
     <zorder>showDebugOutputCheckBox</zorder>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QLineEdit</extends>
   <header>utils/pathchooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="directoryButtonGroup"/>
 </buttongroups>
</ui>
